specification/protocol/otlp.md

```diff
@@ -145,16 +148,47 @@ was not delivered.
 
 #### OTLP/gRPC Response
 
-The server may respond with either a success or an error to the requests.
+The response MUST be the appropriate serialized Protobuf message (see below for
+the specific message to use in the [Success](#success),
+[Partial Success](#partial-success) and [Failure](#failures) cases).
+
+##### Success
 
 The success response indicates telemetry data is successfully processed by the
 server. If the server receives an empty request (a request that does not carry
 any telemetry data) the server SHOULD respond with success.
 
-Success response is returned via
-[Export*ServiceResponse](https://github.com/open-telemetry/opentelemetry-proto)
-message (`ExportTraceServiceResponse` for traces, `ExportMetricsServiceResponse`
-for metrics, `ExportLogsServiceResponse` for logs).
+On success, the server response MUST be
+a Protobuf-encoded [Export*ServiceResponse](https://github.com/open-telemetry/opentelemetry-proto) message
+(`ExportTraceServiceResponse` for traces,
+`ExportMetricsServiceResponse` for metrics and
+`ExportLogsServiceResponse` for logs).
+
+The server SHOULD respond with success no sooner than after successfully
```

Comment:
You mean the size for the response? I couldn't find any orientation for _request size_ at least. I also thought about some constraints around it but I'm not sure what could be a realistic number. 5mb? 10?

Reply:
I also think this will be a sticking point. High volume consumers may want to return immediately and validate async in order to provide low latency communication with OTel agents. Especially if they are consuming a high number of requests and forwarding them somewhere to be validated, processed, stored, etc.

Reply:
Also, the `Failure` section suggests using this [Status](https://cloud.google.com/apis/design/errors) which basically offers a way to add whatever number of error details you want and there's no guidance on how much one should put. Could be an argument on why this should also be unspecified.

Reply:
> High volume consumers may want to return immediately and validate async in order to provide low latency communication with OTel agents. Especially if they are consuming a high number of requests and forwarding them somewhere to be validated, processed, stored, etc.

Good point. But in that case, they can just follow the current Success case. As soon a consumer validates the OTLP request, they can return 200 - OK and the actual processing is done asynchronously. In that case, they can't actually use the partial success and users of such consumer need to rely on something else to know their data is being dropped.

Reply:
That's what I am thinking, but it conflicts (I think) with what you wrote in the spec, right?

Reply:
Ok @pirgeo @dyladan I rephrased the `Success` section a bit. I tried to make it clear now that a server can response with `Success` when:
- It processed the request "right away" and accepted all data
- The server validated the request but will process at a later date (the case where Dan mentioned servers under stress for ex)

This should clarify that a `Success` can be used in both cases. For `Partial Success` then it's clear that it can only be used  when the server knows how many datapoints/logs/spans it processed correctly. This means `Partial Success` can't be used when the server does the process asynchronously